docs/site/Crafting-LoopBack-4.md

```diff
@@ -421,44 +421,50 @@ independent of the nature of such artifacts.
-conclusion is no. Here are the gaps between what Express and LoopBack's needs.
-
-- **Lack of extensibility**.
-
-  Express is a routing and middleware web framework with minimal functionality
-  of its own: An Express application is essentially a series of middleware
-  function calls. For details, see
-  [Using middleware](http://expressjs.com/en/guide/using-middleware.html).
-
-  Express is only extensibile via middleware. It neither exposes a registry nor
-  provides APIs to manage artifacts such as middleware or routers. For its
-  purpose, Express only deals with middleware-like extensions that intercept
-  http requests/responses. LoopBack needs much more extension points and
-  extensions.
-
-- **Lack of composability**.
-
-  Express is not composable. For example, `app.use()` is the only way to
-  register a middleware. The order of middleware is determined by the order of
-  `app.use`. This simplistic approach works for a single monolithic application
-  where all middleware are known and arranged ahead of time. But it does not
-  support the case when middleware from other components need to be added
-  between existing ones. LoopBack had to introduce a phase-based extension and
-  hack Express to provide this capability.
-
-  Express doesn't provide any way to manage dependencies between artifact
-  instances either.
-
-- **Lack of declarative support**.
-
-  In Express, everything is done by JavaScript code as it works exactly as the
-  web site claims: `Fast, unopinionated, minimalist web framework for Node.js`.
-  In contrast, LoopBack is designed to facilitate API creation and composition
-  by conventions and patterns as best practices. More types of constructs are
-  introduced.
+### Why Express Still?
+
+#### Background
+
+LoopBack had always been built on Express so we can leverage the vast community
+and middleware in the Express ecosystem **BUT** it presented some challenges for
+LoopBack. With LoopBack 4 we considered moving away from Express (and even built
+the framework without Express) but eventually circled back to Express because of
+its vast ecosystem.
+
+Some of the gaps between what Express offers and LoopBack's needs are:
+
+- Lack of extensibility
+  > Express is only extensibile via middleware. It neither exposes a registry
+  > nor provides APIs to manage artifacts such as middleware or routers.
+- Lack of composability
+  > Express is not composable. For example, `app.use()` is the only way to
+  > register a middleware. The order of middleware is determined by the order of
+  > `app.use`.
+- Lack of declarative support
+  > In Express, everything is done by JavaScript ... In contrast, LoopBack is
+  > designed to facilitate API creation and composition by conventions and
+  > patterns as best practices.
+
+#### Circling back to Express with a twist
+
+LoopBack does a lot of things but our value add is the ease of use and data
+processing, not middleware and routing so we didn't want to reinvent the wheel
```

Comment:
Routing is the part we want to optimize based on controller metadata. As a result, we minimize the usage of Express routing.